### PR TITLE
JIRA IDVA3-1355 Provide verification details for another psc

### PR DESCRIPTION
**Pull request by AlisonGriffiths**
Creation of a newSubmissionRouter and handler to enable a transaction and submission to be created
Includes unit and integration tests

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/routers/handlers/new-submission/newSubmissionHandler.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    }
+
+    public async createNewSubmission (request: Request, transaction: Transaction): Promise<Resource<PscVerificationResource>> {
+        const companyNumber = request.query.companyNumber as string;
+        const verification: PscVerification = {
+            company_number: companyNumber,
+            // TODO hardcoded value, remove when PSC Id is selected from the List screen
+            psc_appointment_id: "1kdaTltWeaP1EB70SSD9SLmiK5Y"
+        };
```
Comment (hepsimo) [inline src/routers/handlers/new-submission/newSubmissionHandler.ts:43]: Shouldn't this be removed now?

```suggestion
            company_number: companyNumber
        };
```

File: test/routers/handlers/new-submission/newSubmissionHandler.int.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// import { Resource } from "@companieshouse/api-sdk-node";
+// import { Transaction } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
+import { HttpStatusCode } from "axios";
```
Comment (hepsimo) [inline test/routers/handlers/new-submission/newSubmissionHandler.int.ts:3]: Comments can be deleted here

```suggestion
import { HttpStatusCode } from "axios";
```

File: test/routers/handlers/new-submission/newSubmissionHandler.int.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    it("Should redirect with a temporary redirect status code", async () => {
+        const response = await request(app).get(createNewSubmissionUrl).expect(HttpStatusCode.Found);
+    });
+
+    it("Should redirect to the psc_type screen", async () => {
+        const expectedRedirectUrl = `/persons-with-significant-control-verification/transaction/${TRANSACTION_ID}/submission/${PSC_VERIFICATION_ID}/psc-type?lang=en`;
+        const response = await request(app).get(createNewSubmissionUrl).expect("Location", expectedRedirectUrl);
+    });
+});
```
Comment (hepsimo) [inline test/routers/handlers/new-submission/newSubmissionHandler.int.ts:49]: Can be combined into one test if preferred:

```suggestion
    it("Should redirect with a temporary redirect status code to the psc_type screen", async () => {
        const expectedRedirectUrl = `/persons-with-significant-control-verification/transaction/${TRANSACTION_ID}/submission/${PSC_VERIFICATION_ID}/psc-type?lang=en`;
        const response = await request(app).get(createNewSubmissionUrl).expect(HttpStatusCode.Found).expect("Location", expectedRedirectUrl);
    });
});
```

**Review by hepsimo — CHANGES_REQUESTED** [review]
Changed look good, just a little tidying left

File: src/routers/handlers/new-submission/newSubmissionHandler.ts (hunk 1 above)
Comment (AlisonGriffiths) [reply src/routers/handlers/new-submission/newSubmissionHandler.ts]: Now the other PR has been merged in I don't need this hardcoded value anymore. This has been removed

**Review by hepsimo — APPROVED** [review]
All looks good

**Pull request merged**